include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -426,96 +424,76 @@ __parallel_transform_reduce(oneapi::dpl::__internal::__device_backend_tag __back
 {
     auto __n = oneapi::dpl::__ranges::__get_first_range_size(__rngs...);
     assert(__n > 0);
+    using _Size = decltype(__n);
+
+    // Empirically found tuning parameters for typical devices.
+    constexpr _Size __max_iters_per_work_item = 32;
+    constexpr ::std::size_t __max_work_group_size = 256;
+    constexpr ::std::uint8_t __vector_size = 4;
```

Comment:
Does a `__vector_size` of 4 generally perform well for different types or should this be based on the size of the type?

Ex: If `__vector_size = 4` performs best when reducing 32-byte buffers, would `__vector_size = 2` perform best for 64-byte types and `__vector_size = 8` for 16-byte types?

Reply:
This is an interesting question. I will gather some data.

Reply:
4-wide vectors seem to perform best across 16, 32, and 64-bit types on data center GPUs.